Source/ThirdParty/ANGLE/src/common/platform.h

```diff
@@ -127,32 +127,32 @@
 #    include <TargetConditionals.h>
 #    if TARGET_OS_OSX
 #        define ANGLE_PLATFORM_MACOS 1
-#    elif TARGET_OS_IPHONE
-#        define ANGLE_PLATFORM_IOS 1
-#        if TARGET_OS_SIMULATOR
-#            define ANGLE_PLATFORM_IOS_SIMULATOR 1
-#        endif
-#        if TARGET_OS_MACCATALYST
-#            define ANGLE_PLATFORM_MACCATALYST 1
-#        endif
-#    elif TARGET_OS_WATCH
-#        define ANGLE_PLATFORM_WATCHOS 1
-#        if TARGET_OS_SIMULATOR
-#            define ANGLE_PLATFORM_IOS_SIMULATOR 1
-#        endif
-#    elif TARGET_OS_TV
-#        define ANGLE_PLATFORM_APPLETV 1
-#        if TARGET_OS_SIMULATOR
-#            define ANGLE_PLATFORM_IOS_SIMULATOR 1
+#    else
+#        define ANGLE_PLATFORM_APPLE_EMBEDDED 1
+#        if TARGET_OS_IOS
+#            define ANGLE_PLATFORM_IOS 1
+#            if TARGET_OS_MACCATALYST
+#                define ANGLE_PLATFORM_MACCATALYST 1
+#            endif
+#        elif TARGET_OS_WATCH
+#            define ANGLE_PLATFORM_WATCH 1
+#        elif TARGET_OS_TV
+#            define ANGLE_PLATFORM_APPLETV 1
 #        endif
 #    endif
+#
+#    if TARGET_OS_SIMULATOR
+#        define ANGLE_PLATFORM_IOS_SIMULATOR 1
+#    endif
+#
 #    // This might be useful globally. At the moment it is used
 #    // to differentiate MacCatalyst on Intel and Apple Silicon.
 #    if defined(__arm64__) || defined(__aarch64__)
 #        define ANGLE_CPU_ARM64 1
 #    endif
 #    // EAGL should be enabled on iOS, but not Mac Catalyst unless it is running on Apple Silicon.
-#    if (defined(ANGLE_PLATFORM_IOS) && !defined(ANGLE_PLATFORM_MACCATALYST)) || \
+#    if ((defined(ANGLE_PLATFORM_IOS) || defined(ANGLE_PLATFORM_WATCH) \
+        || defined(ANGLE_PLATFORM_APPLETV)) && !defined(ANGLE_PLATFORM_MACCATALYST)) || \
         (defined(ANGLE_PLATFORM_MACCATALYST) && defined(ANGLE_CPU_ARM64))
```

Comment:
The combination `defined(ANGLE_PLATFORM_MACCATALYST) && defined(ANGLE_CPU_ARM64)` appears through out. Should it have a short hand define?